### ISSUE TITLE
Deliver to App Store fails on missing "whatsNew" attribute for app updates

### ISSUE THREAD

**Issue by rubenstolk** [fastlane/fastlane#17318]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description
Sending an app update using `Deliver` fails on:
```
The request cannot be fulfilled because of the state of another resource. - Submit for review errors found.
The provided entity is missing a required attribute - You must provide a value for the attribute 'whatsNew' with this request
```

##### Command executed
`Deliver` using:

`Deliverfile`
```
release_notes({
  'default' => "- Improved number formatting including thousands separator"
})
```

##### Complete output when running fastlane, including the stack trace and command used

```
Traceback (most recent call last):
	34: from /usr/local/bin/fastlane:23:in `<main>'
	33: from /usr/local/bin/fastlane:23:in `load'
	32: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.160.0/bin/fastlane:23:in `<top (required)>'
	31: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.160.0/fastlane/lib/fastlane/cli_tools_distributor.rb:119:in `take_off'
	30: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.160.0/fastlane/lib/fastlane/commands_generator.rb:41:in `start'
	29: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.160.0/fastlane/lib/fastlane/commands_generator.rb:352:in `run'
	28: from /Library/Ruby/Gems/2.6.0/gems/commander-fastlane-4.4.6/lib/commander/delegates.rb:15:in `run!'
	27: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.160.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:76:in `run!'
	26: from /Library/Ruby/Gems/2.6.0/gems/commander-fastlane-4.4.6/lib/commander/runner.rb:476:in `run_active_command'
	25: from /Library/Ruby/Gems/2.6.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:153:in `run'
	24: from /Library/Ruby/Gems/2.6.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:178:in `call'
	23: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.160.0/fastlane/lib/fastlane/commands_generator.rb:108:in `block (2 levels) in run'
	22: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.160.0/fastlane/lib/fastlane/command_line_handler.rb:36:in `handle'
	21: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.160.0/fastlane/lib/fastlane/lane_manager.rb:47:in `cruise_lane'
	20: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.160.0/fastlane/lib/fastlane/runner.rb:45:in `execute'
	19: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.160.0/fastlane/lib/fastlane/runner.rb:45:in `chdir'
	18: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.160.0/fastlane/lib/fastlane/runner.rb:49:in `block in execute'
	17: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.160.0/fastlane/lib/fastlane/lane.rb:33:in `call'
	16: from Fastfile:66:in `block (2 levels) in parsing_binding'
	15: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.160.0/fastlane/lib/fastlane/fast_file.rb:159:in `method_missing'
	14: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.160.0/fastlane/lib/fastlane/runner.rb:157:in `trigger_action_by_name'
	13: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.160.0/fastlane/lib/fastlane/runner.rb:229:in `execute_action'
	12: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.160.0/fastlane/lib/fastlane/runner.rb:229:in `chdir'
	11: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.160.0/fastlane/lib/fastlane/runner.rb:255:in `block in execute_action'
	10: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.160.0/fastlane/lib/fastlane/actions/actions_helper.rb:50:in `execute_action'
	 9: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.160.0/fastlane/lib/fastlane/runner.rb:263:in `block (2 levels) in execute_action'
	 8: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.160.0/fastlane/lib/fastlane/actions/upload_to_app_store.rb:17:in `run'
	 7: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.160.0/deliver/lib/deliver/runner.rb:64:in `run'
	 6: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.160.0/deliver/lib/deliver/runner.rb:194:in `submit_for_review'
	 5: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.160.0/deliver/lib/deliver/submit_for_review.rb:26:in `submit!'
	 4: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.160.0/spaceship/lib/spaceship/connect_api/models/app_store_version.rb:161:in `create_app_store_version_submission'
	 3: from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/forwardable.rb:230:in `post_app_store_version_submission'
	 2: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.160.0/spaceship/lib/spaceship/connect_api/tunes/tunes.rb:860:in `post_app_store_version_submission'
	 1: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.160.0/spaceship/lib/spaceship/connect_api/api_client.rb:124:in `post'
/Library/Ruby/Gems/2.6.0/gems/fastlane-2.160.0/spaceship/lib/spaceship/connect_api/api_client.rb:188:in `handle_response': [!] The request cannot be fulfilled because of the state of another resource. - Submit for review errors found. (Spaceship::UnexpectedResponse)
The provided entity is missing a required attribute - You must provide a value for the attribute 'whatsNew' with this request
```

#### fastlane env
```
<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                                          |
| --------------------------- | -------------------------------------------------------------- |
| OS                          | 10.15.6                                                        |
| Ruby                        | 2.6.3                                                          |
| Bundler?                    | false                                                          |
| Git                         | git version 2.24.3 (Apple Git-128)                             |
| Installation Source         | /usr/local/bin/fastlane                                        |
| Host                        | Mac OS X 10.15.6 (19G2021)                                     |
| Ruby Lib Dir                | /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib |
| OpenSSL Version             | LibreSSL 2.8.3                                                 |
| Is contained                | false                                                          |
| Is homebrew                 | false                                                          |
| Is installed via Fabric.app | false                                                          |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/                    |
| Xcode Version               | 12.0                                                           |


### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE | en_US.UTF-8 | ✅ |


### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
fastlane_version "2.157.2"

default_platform :ios

platform :ios do
  desc "App Store Distribution"
  lane :store do
    match(
      type: "appstore",
      readonly: true,
      keychain_name: "XXXX",
      keychain_password: "XXXX"
    )
    update_project_provisioning(
      xcodeproj: "App/App.xcodeproj",
      profile: ENV['sigh_' + CredentialsManager::AppfileConfig.try_fetch_value(:app_identifier) + '_appstore_profile-path'],
      target_filter: "App",
      build_configuration: "Release",
      code_signing_identity: "Apple Distribution: " + ENV['APPLE_ITC_TEAM_NAME']
    )
    update_project_team(
      path: "App/App.xcodeproj",
      teamid: ENV['APPLE_TEAM_ID']
    )
    gym(workspace: "App/App.xcworkspace")
    deliver(
      force: true,
      submit_for_review: true
    )
  end
end

```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
app_identifier "xxx.xxx.xxx"
apple_id "xxx@xxxx.xx"
team_id "XXXX"
itc_team_name "XXXX"

```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.160.0 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.3.0        |
| slack-notifier            | 2.3.2        |
| rouge                     | 2.0.7        |
| xcpretty                  | 0.3.0        |
| terminal-notifier         | 2.0.0        |
| unicode-display_width     | 1.7.0        |
| terminal-table            | 1.8.0        |
| plist                     | 3.5.0        |
| addressable               | 2.7.0        |
| multipart-post            | 2.0.0        |
| word_wrap                 | 1.0.0        |
| tty-cursor                | 0.7.1        |
| tty-spinner               | 0.9.3        |
| babosa                    | 1.0.3        |
| colored                   | 1.2          |
| highline                  | 1.7.10       |
| commander-fastlane        | 4.4.6        |
| unf_ext                   | 0.0.7.7      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20190701 |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| faraday                   | 1.0.1        |
| faraday_middleware        | 1.0.0        |
| gh_inspector              | 1.1.3        |
| mini_magick               | 4.10.1       |
| rubyzip                   | 2.3.0        |
| security                  | 0.1.3        |
| xcpretty-travis-formatter | 1.0.0        |
| naturally                 | 2.2.0        |
| simctl                    | 1.6.8        |
| uber                      | 0.1.0        |
| declarative-option        | 0.1.0        |
| representable             | 3.0.4        |
| retriable                 | 3.1.2        |
| mini_mime                 | 1.0.2        |
| signet                    | 0.14.0       |
| httpclient                | 2.8.3        |
| google-api-client         | 0.38.0       |
| nanaimo                   | 0.3.0        |
| colored2                  | 3.1.2        |
| claide                    | 1.0.3        |
| CFPropertyList            | 3.0.2        |
| atomos                    | 0.1.3        |
| xcodeproj                 | 1.18.0       |
| public_suffix             | 4.0.5        |
| tty-screen                | 0.8.1        |
| excon                     | 0.76.0       |
| fastimage                 | 2.2.0        |
| json                      | 2.3.1        |
| dotenv                    | 2.7.6        |
| bundler                   | 2.1.4        |
| jwt                       | 2.2.2        |
| declarative               | 0.0.20       |
| multi_json                | 1.15.0       |
| os                        | 1.1.1        |
| memoist                   | 0.16.2       |
| googleauth                | 0.13.1       |
| rake                      | 13.0.1       |
| digest-crc                | 0.6.1        |
| google-cloud-errors       | 1.0.1        |
| google-cloud-env          | 1.3.3        |
| google-cloud-core         | 1.5.0        |
| google-cloud-storage      | 1.28.0       |
| emoji_regex               | 3.0.0        |
| aws-eventstream           | 1.1.0        |
| aws-sigv4                 | 1.2.2        |
| aws-partitions            | 1.363.0      |
| jmespath                  | 1.4.0        |
| aws-sdk-core              | 3.105.0      |
| aws-sdk-kms               | 1.37.0       |
| aws-sdk-s3                | 1.79.1       |
| forwardable               | 1.2.0        |
| logger                    | 1.3.0        |
| stringio                  | 0.0.2        |
| ipaddr                    | 1.2.2        |
| openssl                   | 2.1.2        |
| ostruct                   | 0.1.0        |
| strscan                   | 1.0.0        |
| date                      | 2.0.0        |
| fileutils                 | 1.1.0        |
| etc                       | 1.0.1        |
| io-console                | 0.4.7        |
| zlib                      | 1.0.0        |
| libxml-ruby               | 3.1.0        |
| rexml                     | 3.1.9        |
| psych                     | 3.1.0        |
| mutex_m                   | 0.1.0        |
| webrick                   | 1.4.2        |
</details>

*generated on:* **2020-09-24**
</details>
```

**Comment by max-ott** [comment]
@joshdholtz Did you change the name of the `whats_new`/`release_ note` property?

**Comment by joshdholtz** [comment]
@max-ott The App Store Connect API has the name as `whats_new` but `release_notes` should map back into that! I will take a look right now

**Comment by joshdholtz** [comment]
@rubenstolk Are you setting which languages you are using in your `Deliverfile` or in your `metadata/` directory at all?

http://docs.fastlane.tools/actions/deliver/#default-values says... 👇 [

>In order to use default, you will need to tell deliver which languages your app uses. You can do this in either of two ways:
>
>Create the folders named with the language in the metadata folder (i.e. fastlane/metadata/en-US or fastlane/metadata/de-DE)
>Add the following to your Deliverfile languages(['en-US','de-DE']) 

Just want to make sure you have either of these set!

**Comment by Tapchicoma** [comment]
We are facing the same problem. In our case `release_notes.txt` are placed inside `metadata/` folder for each supported language + `default`.

**Comment by jamesmcmenamin** [comment]
This affects Azure > Apple Store as well as AppCenter > Apple Store (in app center you can choose input changelog notes or txt file)

**Comment by mathaeus** [comment]
Any update on that? :)

We didn't change anything about our setup and it started to fail last week as my colleague mentioned [above](https://github.com/fastlane/fastlane/issues/17318#issuecomment-702153063).

<img width="396" alt="Screenshot 2020-10-09 at 14 41 33" src="https://user-images.githubusercontent.com/5389635/95590252-bcb20c00-0a3d-11eb-955f-a0382c311ace.png">

**Comment by jamesmcmenamin** [comment]
Suppose at some point someone should probably pull-down Fastlane and see what they changed the dang var to... cause someone changed something lol

**Comment by rogerluan** [comment]
If it is of any help, I just ran _deliver_ and it picked up my `release_notes.txt` and submitted to the App Store just fine.

My folder structure:

<img width="265" alt="image" src="https://user-images.githubusercontent.com/8419048/95767565-4c98c580-0c8b-11eb-8f4d-2e3fce063dcd.png">

**Comment by ChristopherDrum** [comment]
Just encountered this issue last night with our Japanese release notes. `en-US` and `es-MX` uploaded without error, but `ja` did not. Suspicious part of the log is here:
```
Uploading metadata to App Store Connect for version
Uploading metadata to App Store Connect for localized version 'en-US'
Uploading metadata to App Store Connect for localized version 'es-MX'
```
The line which appears to correspond to the `ja` version looks malformed?

We have three localizations and the logs before this show proper handling of all three release_notes.txt files up to this point.
Fastlane metadata folder is properly formed with .txt files in each localization, the same as we've been doing for months now.

**Comment by mathaeus** [comment]
@ChristopherDrum unfortunately I don't have the CI job handy to check if we get the same message, but since we also have `ja` I would assume so.

So maybe `deliver` (or something else) expects `ja-JP` instead of `ja`? 
Although [here](https://docs.fastlane.tools/actions/deliver/#available-language-codes) the supported available language codes explicitly list `ja` as well.

**Comment by Green577** [comment]
Thanks for sharing this question ,
i was also looking for the answer keep suggesting such informative post keep sharing with us here so we can also gain some knowledge glad to have you hare [iMessage](https://www.imessageapp.org/)

**Comment by rogerluan** [comment]
Not sure if `ja-JP` is the right code. It's marked as deprecated since September 2015 in _fastlane_ codebase 🤔 `ja` should be the most up-to-date one, although that may have changed in recent Apple API changes

**Comment by mathaeus** [comment]
@rogerluan tbh that was just a wild guess from my side.

@joshdholtz did you have any chance to look into this again?

**Comment by ChristopherDrum** [comment]
@mathaeus I tried `ja-JP` in a test that day, just in case (I had the same wild guess)
The error that came back was from fastlane, not from App Store Connect. So it is possible that an App Store Connect change (the timing of this error is suspiciously close to the iPhone 12 announcement event) needs to be reflected in fastlane? Again, this is all just speculation.
Trying to access the "App Stores and localizations" help to see if there are release notes on this issue, or some documented change, and that link just doesn't open at all (`javascript:void(0)`). So... I'm not sure how to proceed.

<img width="361" alt="Screen Shot 2020-10-16 at 8 57 04 AM" src="https://user-images.githubusercontent.com/320377/96197619-92a4a080-0f8d-11eb-93d0-7812ad0e4597.png">

**Comment by mathaeus** [comment]
Thanks @ChristopherDrum. I experience the same non-helping help link across different browsers :/ 

I don't know about the internals of `fastlane` or `deliver` so eventually somebody with more knowledge needs to help :)

**Comment by mathaeus** [comment]
@ChristopherDrum update from my side. It seems to work again for us 🤷‍♂️ . 

At least I see the `What's New in This Version` field being filled correctly on `App Store Connect`.
```
[09:18:05]: Detected languages: ["de-DE", "default", "en-AU", "en-CA", "en-GB", "en-US", "es-ES", "es-MX", "fr-CA", "fr-FR", "it", "ja", "ko", "pl", "pt-BR", "pt-PT", "ru", "tr"]
[09:18:08]: Cannot find edit app store version... Retrying after 10 seconds (remaining: 4)
[09:18:24]: Will begin uploading metadata for '6.43.0' on App Store Connect
[09:18:24]: Uploading metadata to App Store Connect for version
[09:18:25]: Uploading metadata to App Store Connect for localized version 'it'
[09:18:25]: Uploading metadata to App Store Connect for localized version 'pt-BR'
[09:18:26]: Uploading metadata to App Store Connect for localized version 'en-AU'
[09:18:27]: Uploading metadata to App Store Connect for localized version 'ja'
[09:18:27]: Uploading metadata to App Store Connect for localized version 'fr-CA'
[09:18:28]: Uploading metadata to App Store Connect for localized version 'ru'
[09:18:28]: Uploading metadata to App Store Connect for localized version 'fr-FR'
[09:18:29]: Uploading metadata to App Store Connect for localized version 'tr'
[09:18:30]: Uploading metadata to App Store Connect for localized version 'en-US'
[09:18:30]: Uploading metadata to App Store Connect for localized version 'es-ES'
[09:18:31]: Uploading metadata to App Store Connect for localized version 'pl'
[09:18:32]: Uploading metadata to App Store Connect for localized version 'es-MX'
[09:18:32]: Uploading metadata to App Store Connect for localized version 'en-CA'
[09:18:33]: Uploading metadata to App Store Connect for localized version 'de-DE'
[09:18:33]: Uploading metadata to App Store Connect for localized version 'ko'
[09:18:34]: Uploading metadata to App Store Connect for localized version 'pt-PT'
[09:18:35]: Uploading metadata to App Store Connect for localized version 'en-GB'
```

**Comment by ChristopherDrum** [comment]
@mathaeus Then I wonder why fastlane was skipping the `ja` folder 🤔 
Thanks for the update. I started wondering about `default` when I saw it doesn't specifically mention `localized` in the message. But I also don't have a `default` folder and only have 3 languages so it seemed at first like a one-to-one; three lines in the logs and three languages. 

Thanks for the update and fingers crossed I'll have a good release later today. 🤞

**Comment by ChristopherDrum** [comment]
Found an old .log file which contains the time when release_notes.txt upload success transitioned to failure.

On Sep. 21, v2.157.2 -> success
On Oct. 12, v2.162.0 -> failure (and every version since then)

**Comment by mathaeus** [comment]
@ChristopherDrum that seems to be weird indeed. We also haven't really changed anything about our setup and on Friday it worked. BUT on Friday I executed the according lane locally and not on CI. Although that shouldn't really matter, but who knows. I guess I will see soon, if we keep having the problem on CI.

**Comment by ChristopherDrum** [comment]
OK, I finally had success today. I don't understand why this worked, but it did...
In my `deliver()` parameters for the App Store Connect submission, I had `metadata_dir:` set to the local folder that contained the metadata for each language.

I removed this parameter and let fastlane find the folder on its own. Exact same folder, and I know it was because it found and uploaded every release-notes.txt file for every language except `ja`. So it saw the data, and it even logged `ja` text file to console as existing.

But the real upload skipped `ja` every time.

So if I define `metadata_dir` in the deliver parameters, `ja` is skipped.
If I remove this parameter, `ja` is not skipped.
This is new behavior, somewhere between fastlane v2.157.2 and v2.162.0

**Comment by mathaeus** [comment]
@ChristopherDrum for us it hasn't failed ever since my last comment, and we also have `ja`. So 🤷‍♂️. It stays a mystery.

**Comment by ChristopherDrum** [comment]
Well, the success noted above was short-lived. It worked last week, but this week fastlane decided that the fix wasn't good enough and skipped `ja` release notes again. 🤷 
I did update fastlane to `2.168.0` just before that. I don't know; I'm at a loss now.

**Comment by ChristopherDrum** [comment]
Hmm.. #17713 doesn't seem to apply to my case. I just checked my server and it was (still is) on 2.168.0 though the issue mentioned above is on 2.169.0

**Comment by ChristopherDrum** [comment]
@mathaeus I just noticed something you said
`BUT on Friday I executed the according lane locally and not on CI`

On the day I had success, I had also run the lane locally, not on CI. 
I saw that worked, and thought, "Ah, it's working now!"
So I went into App Store Connect and manually deleted the uploaded release notes, then the CI kicked off and everything uploaded (at least according to the log files) as expected.

Is there a flag or setting in App Store Connect that needs to be "unlocked" (so to speak) for release notes to successfully upload? Or for fastlane to detect the languages in the store? 🤔
